File: Lib/Telerik/Examples/QuickStart/Scheduler/CustomElements/MyElementProvider.vb
Imports Telerik.Examples.WinControls.Scheduler.CustomElements
Imports Telerik.WinControls.UI

Public Class MyElementProvider
    Inherits SchedulerElementProvider
    Public Sub New(scheduler As RadScheduler)
        MyBase.New(scheduler)
    End Sub

    Protected Overrides Function CreateElement(Of T As SchedulerVisualElement)(view As SchedulerView, context As Object) As T
        If GetType(T).Equals(GetType(AppointmentElement)) Then
            Return TryCast(New MyAppointmentElement(Me.Scheduler, view, DirectCast(context, IEvent)), T)
        End If
        If GetType(T).Equals(GetType(SchedulerCellElement)) Then
            Return TryCast(New MySchedulerCellElement(Me.Scheduler, view), T)
        End If

        Return MyBase.CreateElement(Of T)(view, context)
    End Function


    Public Overrides Function CreateRulerPrimitive(area As DayViewAppointmentsArea, timeZone As SchedulerTimeZone) As RulerPrimitive
        Return New MyRulerPrimitive(Me.Scheduler, area)
    End Function
End Class

File: Lib/Telerik/Examples/QuickStart/Scheduler/CustomElements/MyElementProvider.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using Telerik.WinControls.UI;

namespace Telerik.Examples.WinControls.Scheduler.CustomElements
{
    public class MyElementProvider : SchedulerElementProvider
    {
        public MyElementProvider(RadScheduler scheduler)
            : base(scheduler)
        {
        }

        protected override T CreateElement<T>(SchedulerView view, object context)
        {
            if (typeof(T) == typeof(AppointmentElement))
            {
                return new MyAppointmentElement(this.Scheduler, view, (IEvent)context) as T;
            }
            if (typeof(T) == typeof(SchedulerCellElement))
            {
                return new MySchedulerCellElement(this.Scheduler, view) as T;
            }

            return base.CreateElement<T>(view, context);
        }

        public override RulerPrimitive CreateRulerPrimitive(DayViewAppointmentsArea area, SchedulerTimeZone timeZone)
        {
            return new MyRulerPrimitive(this.Scheduler, area);
        }
    }
}
